src/common/meta/src/wal/kafka/topic_manager.rs

```diff
@@ -117,14 +124,20 @@ impl TopicManager {
             .await
             .with_context(|_| BuildKafkaClientSnafu {
                 broker_endpoints: self.config.broker_endpoints.clone(),
-            })?
+            })?;
+
+        let control_client = client
             .controller_client()
             .context(BuildKafkaCtrlClientSnafu)?;
 
         // Try to create missing topics.
         let tasks = to_be_created
             .iter()
-            .map(|i| self.try_create_topic(&topics[*i], &client))
+            .map(|i| async {
+                self.try_create_topic(&topics[*i], &control_client).await?;
+                self.try_append_noop_record(&topics[*i], &client).await?;
```

Comment:
What if the topic creation is success but the append is failed here? Or What if the append returns failure because of some network errors but it is actually succeed in kafka site? Will the append be executed the next time and thus two noop records are appended? There might be other situations that could make inconsistency, I think what we ought to do here is to make the "create and append" a transaction. But can we?

Reply:
Perhaps we should change the behavior of `last_entry_id`? cc @evenyag

Reply:
> What if the topic creation is success but the append is failed here? Or What if the append returns failure because of some network errors but it is actually succeed in kafka site? Will the append be executed the next time and thus two noop records are appended? There might be other situations that could make inconsistency, I think what we ought to do here is to make the "create and append" a transaction. But can we?

Yes, we need to leave an issue for this. We must ensure the topic offset is greater than 0.